Replace usage of rand_r with nrand48

dnewhall opened the pull request:
This replaces the call to `rand_r` with `nrand48` and changes `rand_seed` to be an array instead of an unsigned int.

With `rand_r` being deprecated in POSIX 2024, the best replacement is `nrand48`, which stores its intermediate values as a 48-bit integer in an array of shorts instead of an unsigned int and returns a 32-bit value not constrained by `RAND_MAX`.

`nrand48` has been in the POSIX spec. since 2004 and on the large UNIX systems since SVr4. Only things that could be affected are BSDs older than ~2004, but those are presumably old enough that we can ignore them.

`bin/package test` returned with all tests passing (particularly tests/subshell.sh and tests/variables.sh) and monkey testing shows getting and setting `RANDOM` works.

dnewhall committed:
Replaced usage of rand_r with nrand48

McDutchie committed:
libast: cleanup: remove unused astcopy function

It has been unused in the entire history of the ast-open-archive
repo as well as ours.

Also stop generating the public ast_mmap.h header, which is also
unused (it's a copy of FEATURE/mmap which is used internally by
libast).

McDutchie commented:
Sorry for the delay in getting to this, I've been busy elsewhere. It looks good to me at first glance. Hopefully I'll find some time to take a closer look and test it in the next few days.

McDutchie committed:
Merge branch 'dev' into dnewhall

McDutchie committed:
housekeeping

Bump version date
Update ANNOUNCE and NEWS

Undo whitespace fixes in function declarations (I'll save those for an
accumulated cleanups commit that will also be backported to 1.0)
